pkg/services/publicdashboards/service/service.go

```diff
@@ -362,6 +362,11 @@ func (pd *PublicDashboardServiceImpl) DeleteByDashboard(ctx context.Context, das
 		return nil
 	}
 
+	// validate the public dashboard belongs to the dashboard
+	if pubdash.DashboardUid != dashboard.UID {
```

Comment:
This method `deleteByDashboard` retrieves both the dashboard and the pubdash from the database so this validation will never fail.

Reply:
Hello @evictorero, I made the code changes. Let me know if these are okay, will redo the UT then. Also, is it okay to change the test cases of the existing UT from a for loop into individual test cases?